### PR TITLE
Text Recognition 0016

### PR DESCRIPTION
**Pull request by morkovka1337**
This PR adds important files for the text recognition 0016 model. No changes in demo or AC are needed.

### CODE REVIEW

**Comment by vladimir-dudnik** [comment]
@morkovka1337 please add model to relevant demo models.lst file and to relevant test cases (although leave these changes commented, to not break precommit checks until model is available)

cc @Wovchena

**Comment by morkovka1337** [comment]
@vladimir-dudnik there is regexp `text-recognition-????-encoder` & `text-recognition-????-decoder` in the models.lst, do I still have to add anything to this file?

**Comment by vladimir-dudnik** [comment]
@morkovka1337 ok, no need to add to models.lst but still need a specific test case in cases.py

**Comment by morkovka1337** [comment]
> @morkovka1337 ok, no need to add to models.lst but still need a specific test case in cases.py

Test added (commented)

**Comment by Wovchena** [comment]
@morkovka1337, with TODO or FIXME mark, please. https://github.com/openvinotoolkit/open_model_zoo/blob/develop/ci/check-release-readiness.py#L56

File: models/intel/text-recognition-0016/composite-model.yml [hunk 1]
```diff
@@ -0,0 +1 @@
+task_type: token_recognition
```
Comment (vladimir-dudnik) [inline models/intel/text-recognition-0016/composite-model.yml:1]: old text-recognition models (0012/0014) have type optical_character_recognition but new ones, 0015/0016 got this token_recognition. Is there any special reason for that difference?

**Review by Wovchena — APPROVED** [review]

File: models/intel/text-recognition-0016/composite-model.yml (hunk 1 above)
Comment (morkovka1337) [reply models/intel/text-recognition-0016/composite-model.yml:1]: I just copypasted the 0015 model. Do not remember, if it was me who added this `token_recognition` type.

Comment (Wovchena) [reply models/intel/text-recognition-0016/composite-model.yml:1]: try `git blame` :)

Comment (morkovka1337) [reply models/intel/text-recognition-0016/composite-model.yml:1]: Well, seems like it was me. I can change it to `optical_character_recognition` if it is needed.

Comment (vladimir-dudnik) [reply models/intel/text-recognition-0016/composite-model.yml:1]: @morkovka1337 being consistent is always better, otherwise we easily fall into chaos

Comment (morkovka1337) [reply models/intel/text-recognition-0016/composite-model.yml]: should I also change the corresponding value in 0015th model?

Comment (vladimir-dudnik) [reply models/intel/text-recognition-0016/composite-model.yml]: @morkovka1337 yes, please

Comment (morkovka1337) [reply models/intel/text-recognition-0016/composite-model.yml]: Done

File: models/intel/text-recognition-0016/accuracy-check.yml [hunk 2]
```diff
@@ -0,0 +1,108 @@
+evaluations:
+  - name: text-recognition-0016
+    module: custom_evaluators.custom_text_recognition_evaluator.TextRecognitionWithAttentionEvaluator
+    module_config:
+      model_type: SequentialTextRecognitionModel
+      lowercase: true
+      network_info:
+        max_seq_len: "24"
```
Comment (eaidova) [inline models/intel/text-recognition-0016/accuracy-check.yml:8]: after merging https://github.com/openvinotoolkit/open_model_zoo/pull/2880 config should be updated
cc @Anna-Mironova

File: models/intel/text-recognition-0016/README.md [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+The text-recognition-0016-encoder model is a ResNeXt-101 like backbone with TPS network and convolutional encoder part of the text recognition.
+
+| Metric  | Value |
+| ------- | ----- |
+| GFlops  | 9.27  |
+| MParams | 88.1  |
+
+### Inputs
+
+Image, name: `imgs`, shape: `1, 1, 64, 256` in the `1, C, H, W` format, where:
```
Comment (eaidova) [inline models/intel/text-recognition-0016/README.md:40]:
```suggestion
Image, name: `imgs`, shape: `1, 1, 64, 256` in the `N, C, H, W` format, where:
```

File: models/intel/text-recognition-0016/README.md [hunk 4]
```diff
@@ -0,0 +1,85 @@
...
+
+- `C` - number of channels
+- `H` - image height
+- `W` - image width
```
Comment (eaidova) [inline models/intel/text-recognition-0016/README.md:44]:
```suggestion
- `B` - batch size 
- `C` - number of channels
- `H` - image height
- `W` - image width
```

File: models/intel/text-recognition-0016/README.md [hunk 5]
```diff
@@ -0,0 +1,85 @@
...
+### Inputs
+
+1.	Name: `decoder_input`, shape: `1`. Previous predicted letter.
+2.	Name: `features`, shape: `1, 36, 1024`. Encoded features.
+3.	Name: `hidden`, shape: `1, 1, 1024`. Current state of the decoder.
+
+### Outputs
+
+1.	Name: `decoder_hidden`, shape: `1, 1, 1024`. Current context state of the LSTM cell.
+2.	Name: `decoder_output`, shape: `1, 40`. Classification confidence scores in the [0, 1] range
```
Comment (eaidova) [inline models/intel/text-recognition-0016/README.md:70]: what about specification decoding symbols more explicit then just key for demo?

File: models/intel/text-recognition-0016/README.md [hunk 6]
```diff
@@ -0,0 +1,85 @@
...
+
+
+### Outputs
+
+1.	Name: `decoder_hidden`, shape: `1, 1, 1024`. Initial context state of the GRU cell.
+2.	Name: `features`, shape: `1, 36, 1024`. Features from encoder part of text recognition head.
+
+## Decoder model specification
+
+The text-recognition-15-decoder model is a GRU based decoder with 2d attention module.
```
Comment (eaidova) [inline models/intel/text-recognition-0016/README.md:54]:
```suggestion
The text-recognition-0016-decoder model is a GRU based decoder with 2d attention module.
```

File: models/intel/text-recognition-0016/README.md [hunk 7]
```diff
@@ -0,0 +1,85 @@
...
+
+| Metric                                         | Value              |
+| ---------------------------------------------- | ------------------ |
+| Accuracy on the alphanumeric subset of ICDAR13 | 0.9685             |
+| Accuracy on the alphanumeric subset of ICDAR03 | 0.9712             |
+| Accuracy on the alphanumeric subset of ICDAR15 | 0.8675             |
+| Accuracy on the alphanumeric subset of SVT     | 0.9474             |
+| Accuracy on the alphanumeric subset of IIIT5K  | 0.9347             |
+| Text location requirements                     | Tight aligned crop |
+| Source framework                               | PyTorch\*          |
```
Comment (eaidova) [inline models/intel/text-recognition-0016/README.md:27]: probably Source Framework should be placed before accuracy numbers as it is more important and general info

**Review by eaidova — APPROVED** [review]

**Comment by eaidova** [comment]
@morkovka1337 please in the next time use own fork for creation branch instead of main OMZ repo

**Review by vladimir-dudnik — APPROVED** [review]

**Pull request merged**